internal/handlers/bot.go

```diff
@@ -73,28 +86,113 @@ type IncomingEvent struct {
 	Event    Event  `json:"event"`
 }
 
+func loadRichMessage() (json.RawMessage, error) {
+	data, err := os.ReadFile("rich_message.json")
+	if err != nil {
+		return nil, err
+	}
+	return json.RawMessage(data), nil
+}
+
 func (h *WebhookHandler) Reply(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusAccepted)
-
 	var webhook Webhook
 
 	if err := json.NewDecoder(r.Body).Decode(&webhook); err != nil {
 		log.Printf("Error decoding webhook JSON: %v", err)
 		return
 	}
-
 	defer r.Body.Close()
 
 	log.Printf("Received webhook: %s, payload %s", webhook.Action, string(webhook.Payload))
 
 	switch {
 	case webhook.Action == "incoming_event":
-		log.Printf("Incoming envet action")
+		if err := h.handleIncomingEvent(webhook); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 	case webhook.Action == "incoming_chat":
-		log.Printf("Incoming chat action")
+		if err := h.handleIncomingChat(webhook); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 	default:
 		log.Printf("Webhook not recognized: %v", webhook)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *WebhookHandler) handleIncomingEvent(webhook Webhook) error {
+	var incomingEvent IncomingEvent
+	if err := json.Unmarshal(webhook.Payload, &incomingEvent); err != nil {
+		return fmt.Errorf("failed to unmarshall incoming event payload: %w", err)
+	}
+	token, exists := h.botStore.GetBotToken(webhook.OrganizationID)
+	if !exists {
+		return fmt.Errorf("bot token does not exists for organizationID: %s", webhook.OrganizationID)
+	}
+	var postback Postback
+	if err := json.Unmarshal(incomingEvent.Event.Postback, &postback); err != nil {
+		return fmt.Errorf("failed to unmarshal postback: %w", err)
+	}
+	log.Printf("Postback ID- %s", postback.ID)
+	switch postback.ID {
+	case "transfer_to_agent":
+		return h.handleTransferToAgent(incomingEvent, token)
+	case "continue_chat_with_bot":
+		return h.handleContinueWithBot(incomingEvent, token)
```

Comment:
that's purely my opinion, but I'd reserve `handle...` name for the type (action) of incoming webhook, to not mess up these two separate things. These 2 methods could be named just `transferToAgent` & `continueWithBot`